.github/scripts/run-fips-ut.sh

```diff
@@ -6,7 +6,11 @@ fips-mode-setup --is-enabled
 if [ $? -ne 0 ]; then
   exit 1
 fi
-echo "fips.provider.7=XMLDSig" >>/etc/alternatives/java_sdk_17/conf/security/java.security
+JAVA_SECURITY_FILE="/etc/alternatives/java_sdk_17/conf/security/java.security"
+
+echo "fips.provider.7=XMLDSig" >> "${JAVA_SECURITY_FILE}"
+sed -i 's/securerandom.strongAlgorithms=NativePRNGBlocking:SUN,DRBG:SUN/securerandom.strongAlgorithms=PKCS11:SunPKCS11-NSS-FIPS/g' "${JAVA_SECURITY_FILE}"
```

Comment:
I think we cannot bypass this change and we should update documentation. For what I see there are change in [bc-fips 1.0.2.4](https://www.bouncycastle.org/download/bouncy-castle-java-fips/), it says:

> Identification and use of system entropy source optimizations (elimination of possible exception throwing, configuration for thread local SecureRandom provider) All secret key and private key types now provide support for Destroyable. Hex.decodeStrict() added to Hex encoder.

For what I see, now the fips is initialized to use the `SecureRandom.getInstanceStrong` to get the random. RHEL has this [bug](https://issues.redhat.com/browse/RHEL-3478). And bc-fips, if that method fails, uses directly the sun implementation `new sun.security.provider.SecureRandom()` which also fails with `cannot access class sun.security.provider.SecureRandom (in module java.base) because module java.base does not export sun.security.provider to unnamed module` in jdk 17 or 21. So I think we should document this and say it's needed now.

Reply:
I was wrong, the only difference is that now we need to maintain the `Security.setProperty("securerandom.strongAlgorithms", usedAlg);` all the time (previously it was removed in in the finally). Let's see the tests in my branch.